### PR TITLE
Norman

### PR DESCRIPTION
**Pull request by normanmukama**
This is the updated code of our project.  #FinishBefore2024

### CODE REVIEW

File: app/Config/Routes.php [hunk 1]
```diff
@@ -79,9 +81,14 @@
...
 $routes->post('fetchjobtype', 'JobtypeController::fetchjobtype');
 
+// Working with payroll
+$routes->post('fetchmanagepayroll', 'ManagepayrollController::fetchmanagepayroll');
+
+
+
 /*
  * --------------------------------------------------------------------
  * Additional Routing
```
Comment (Leon12-aaryson) [inline app/Config/Routes.php:94]: configure routes

File: app/Controllers/BaseController.php [hunk 2]
```diff
@@ -35,7 +35,7 @@ abstract class BaseController extends Controller
      *
      * @var array
      */
-    protected $helpers = ['url'];
+    protected $helpers = ['url','customized'];
 
     /**
      * Be sure to declare properties for any property fetch you initialized.
```
Comment (Leon12-aaryson) [inline app/Controllers/BaseController.php:41]: customizing helpers

File: app/Controllers/Home.php [hunk 3]
```diff
@@ -1,10 +1,20 @@
...
     public function __construct()
     {
-        helper(['html','form']);
+        // Loading the database service via dependency injection
+        $this->db = \Config\Database::connect();
+        $this->payroll = new PayrollModel();
+        helper(['html', 'form']);
     }
 
     public function index(): string
```
Comment (aaron-oluk) [inline app/Controllers/Home.php:20]: adding controllers for payroll

File: app/Controllers/BaseController.php (hunk 2 above)
Comment (aaron-oluk) [inline app/Controllers/BaseController.php:41]: customizing the helpers

File: app/Config/Routes.php (hunk 1 above)
Comment (aaron-oluk) [inline app/Config/Routes.php:94]: working on routes

**Review by aaron-oluk — APPROVED** [review]
new updates to controllers and routes

**Pull request merged**